skodaconnect/connection.py

```diff
@@ -1093,21 +1093,43 @@ async def getTripStatistics(self, vin):
                     f'fs-car/bs/tripstatistics/v1/{BRAND}/{COUNTRY}/vehicles/{vin}/tripdata/longTerm?newest'
                 )
             )
+            cyclic = await self.get(
+                urljoin(
+                    self._session_auth_ref_url[vin],
+                    f'fs-car/bs/tripstatistics/v1/{BRAND}/{COUNTRY}/vehicles/{vin}/tripdata/cyclic?newest'
+                )
+            )
             data = {}
-            if short.get('tripData', False) and long.get('tripData', False):
+            if short.get('tripData', False) and long.get('tripData', False) and cyclic.get('tripData', False):
                 data['tripstatistics'] = short.get('tripData', {})
                 data['longtermstatistics'] = long.get('tripData', {})
+                data['cyclicstatistics'] = cyclic.get('tripData', {})
                 return data
-            elif short.get('tripData', False):
+            elif short.get('tripData', False) and long.get('tripData', False):
+                data['tripstatistics'] = short.get('tripData', {})
+                data['longtermstatistics'] = long.get('tripData', {})
+                return data
+            elif short.get('tripData', False) and cyclic.get('tripData', False):
                 data['tripstatistics'] = short.get('tripData', {})
+                data['cyclicstatistics'] = cyclic.get('tripData', {})
+                return data
+            elif cyclic.get('tripData', False) and long.get('tripData', False):
+                data['cyclicstatistics'] = cyclic.get('tripData', {})
+                data['longtermstatistics'] = long.get('tripData', {})
                 return data
+            elif short.get('tripData', False):
+                data['tripstatistics'] = short.get('tripData', {})
             elif long.get('tripData', False):
                 data['longtermstatistics'] = long.get('tripData', {})
+            elif cyclic.get('tripData', False):
+                data['cyclicstatistics'] = cyclic.get('tripData', {})
                 return data
             elif short.get('status_code', False):
                 _LOGGER.warning(f'Could not fetch trip statistics, HTTP status code: {short.get("status_code")}')
             elif long.get('status_code', False):
-                _LOGGER.warning(f'Could not fetch trip statistics, HTTP status code: {long.get("status_code")}')
+                _LOGGER.warning(f'Could not fetch longtermin trip statistics, HTTP status code: {long.get("status_code")}')
+            elif cyclic.get('status_code', False):
+                _LOGGER.warning(f'Could not fetch cyclic trip statistics, HTTP status code: {cyclic.get("status_code")}')
             else:
                 _LOGGER.info(f'Unhandled error while trying to fetch trip statistics')
```

Comment:
Consider reducing and simplifying this huge if statement to something like:

```python
        data = {}
        try:
            ...
            if short.get('tripData', False):
                data['tripstatistics'] = short.get('tripData', {})
            else:
                _LOGGER.warning(f'Could not fetch trip statistics, HTTP status code: {short.get("status_code", "none")}')
            if long.get('tripData', False):
                data['longtermstatistics'] = long.get('tripData', {})
            else:
                _LOGGER.warning(f'Could not fetch longtermin trip statistics, HTTP status code: {long.get("status_code", "none")}')
            if cyclic.get('tripData', False):
                data['cyclicstatistics'] = cyclic.get('tripData', {})
            else:
                _LOGGER.warning(f'Could not fetch cyclic trip statistics, HTTP status code: {cyclic.get("status_code", "none")}')
        except Exception as error:
            _LOGGER.warning(f'Could not fetch trip statistics, error: {error}')
        return data


```

Reply:
I get your point. Code could be easier, cleaner and shorter.
But there are dozens of similiar code-blocks with `if .. elif .. elif .. ` and I tried not to change the style of code.
I could make another commit I you like (already prepared).